predicates/interval/interval.go

```diff
@@ -0,0 +1,168 @@
+	example2: Path("/zalando") && Between(1451642400, 1454320800) -> "https://www.zalando.de";
+
+	example3: Path("/zalando") && Before("2016-02-01T12:00:00+02:00") -> "https://www.zalando.de";
+	example4: Path("/zalando") && Before(1454320800) -> "https://www.zalando.de";
+
+	example3: Path("/zalando") && After("2016-01-01T12:00:00+02:00") -> "https://www.zalando.de";
+	example4: Path("/zalando") && After(1451642400) -> "https://www.zalando.de";
+
+*/
+package interval
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/zalando/skipper/routing"
+)
+
+type intervalType int
+
+const (
+	between intervalType = iota
+	before
+	after
+)
+
+var InvalidArgsError = errors.New("invalid arguments")
+
+type spec struct {
+	typ intervalType
+}
+
+type predicate struct {
+	typ   intervalType
+	begin time.Time
+	end   time.Time
+	time  func() time.Time
+}
+
+// Creates Between predicate.
+func NewBetweenPredicate() routing.PredicateSpec { return &spec{between} }
+
+// Creates Before predicate.
+func NewBeforePredicate() routing.PredicateSpec { return &spec{before} }
+
+// Creates After predicate.
+func NewAfterPredicate() routing.PredicateSpec { return &spec{after} }
+
+func (s *spec) Name() string {
+	switch s.typ {
+	case between:
+		return "Between"
+	case before:
+		return "Before"
+	case after:
+		return "After"
+	default:
+		panic("invalid interval predicate type")
+	}
+}
+
+func (s *spec) Create(args []interface{}) (routing.Predicate, error) {
+	switch s.typ {
+	case between:
+		if len(args) != 2 {
+			return nil, InvalidArgsError
+		}
+	default:
+		if len(args) != 1 {
+			return nil, InvalidArgsError
+		}
+	}
+
+	time := func() time.Time {
+		return time.Now()
+	}
+
+	switch s.typ {
+	case between:
+		if begin, end, ok := parseArgs(args[0], args[1]); ok {
+			if begin.Before(end) {
+				return &predicate{s.typ, begin, end, time}, nil
+			}
+		}
+	case before:
+		if end, ok := parseArg(args[0]); ok {
+			return &predicate{typ: s.typ, end: end, time: time}, nil
+		}
+	case after:
+		if begin, ok := parseArg(args[0]); ok {
+			return &predicate{typ: s.typ, begin: begin, time: time}, nil
+		}
+	}
+
+	return nil, InvalidArgsError
+}
+
+func parseArgs(arg1, arg2 interface{}) (time.Time, time.Time, bool) {
+	if begin, ok := parseArg(arg1); ok {
+		if end, ok := parseArg(arg2); ok {
+			return begin, end, true
+		}
+	}
+
+	return time.Time{}, time.Time{}, false
+}
+
+func parseArg(arg interface{}) (time.Time, bool) {
+	switch a := arg.(type) {
+	case string:
+		t, err := time.Parse(time.RFC3339, a)
+		if err == nil {
+			return t, true
+		}
+	case float64:
+		return time.Unix(int64(a), 0), true
+	case int64:
+		return time.Unix(a, 0), true
+	}
+
+	return time.Time{}, false
+}
+
+func (p *predicate) Match(r *http.Request) bool {
+	now := p.time()
+
+	switch p.typ {
+	case between: // Between is inclusive and Before and After are exclusive
```

Comment:
You got it right, I made it my way :) I think it is a bit confusing for non tech people. If upper boundary is excluded then Before and After will be have a bit different. In Before the date will be excluded, because it is the right boundary. In After the date will be included, because it is the left boundary. I made it simpler, all explicitly defined dates are included.